File: api/src/main/java/com/septem1997/api/entity/Barrage.java
package com.septem1997.api.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;
import java.io.Serializable;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
@Entity
@Table
@DynamicUpdate
public class Barrage extends BaseEntity implements Serializable {
    private String content;

//    @JsonBackReference
//    @ManyToOne(fetch = FetchType.LAZY,optional=false)
//    @JoinColumn(name="room_id")
//    private Room room;

    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY,optional=false)
    @JoinColumn(name="user_id")
    private User user;

    private String userNickname;
}

File: backend/src/entity/barrage.ts
import { Entity, Column, PrimaryGeneratedColumn, OneToMany, ManyToOne } from 'typeorm';
import {BaseEntity} from "./baseEntity";
import { RoomTag } from './roomTag';
import { User } from './user';
import { Room } from './room';

@Entity()
export class Barrage extends BaseEntity{

    @Column()
    content: string;

    @Column()
    size: number;

    @Column()
    color: string;

    @ManyToOne(() => User,{
        nullable:true
    })
    creator: User;

    sender:string;

    @ManyToOne(() => Room)
    room: Room;
}
